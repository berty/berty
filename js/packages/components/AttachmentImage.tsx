import React, { useState, useEffect } from 'react'
import { Image, ImageProps, ActivityIndicator, View, TouchableOpacity } from 'react-native'
import { useNavigation } from '@react-navigation/native'

import { useMessengerContext } from '@berty-tech/store'

import { getSource } from './utils'

const AttachmentImage: React.FC<{ cid: string; pressable?: boolean } & Omit<ImageProps, 'source'>> =
	props => {
		const { navigate } = useNavigation()
		const { protocolClient, medias } = useMessengerContext()
		const [source, setSource] = useState('')
		const { cid, ...imageProps } = props
		const mimeType = medias[cid]?.mimeType || 'image/jpeg'

		useEffect(() => {
			if (!protocolClient) {
				return
			}
			let cancel = false
			getSource(protocolClient, cid)
				.then(src => {
					if (!cancel) {
						setSource(`data:${mimeType};base64,${src}`)
					}
				})
				.catch(e => console.error('failed to get attachment image:', e))
			return () => {
				cancel = true
			}
		}, [protocolClient, cid, mimeType])

		if (!source) {
			return (
				<View style={[props.style, { alignItems: 'center', justifyContent: 'center' }]}>
					<ActivityIndicator color='grey' />
				</View>
			)
		}

		return !props.pressable ? (
			<Image source={{ uri: source }} {...imageProps} />
		) : (
			<TouchableOpacity
				onPress={() => {
					navigate('Modals.ImageView', { images: [{ uri: source }], previewOnly: true })
				}}
			>
				<Image source={{ uri: source }} {...imageProps} />
			</TouchableOpacity>
		)
	}

export default AttachmentImage
