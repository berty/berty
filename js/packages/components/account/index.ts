export { StreamProgress } from './StreamProgress'
export { CenteredTextScreen } from './CenteredTextScreen'
