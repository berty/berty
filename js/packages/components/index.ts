export * from './buttons'
export * from './create-group'
export * from './inputs'
