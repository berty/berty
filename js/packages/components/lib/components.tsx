import LibExample from "./Example.tsx"

export const Example = LibExample
