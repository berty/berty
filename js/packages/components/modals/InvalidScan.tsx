import React, { useState } from 'react'
import { View, StyleSheet, TouchableOpacity } from 'react-native'
import { Text, Icon } from 'react-native-ui-kitten'
import { useStyles } from '@berty-tech/styles'
import { useNavigation } from '@react-navigation/native'
import { Messenger } from '@berty-tech/hooks'

//
// Scan Invalid
//

// Styles
const useStylesInvalidScan = () => {
	const [{ width, height, border, text, padding, margin }] = useStyles()
	return {
		header: [width(120), height(120), border.radius.scale(60)],
		dismissButton: [
			border.color.light.grey,
			border.scale(2),
			border.radius.small,
			margin.top.scale(50),
			padding.left.small,
			padding.right.medium,
			padding.top.small,
			padding.bottom.small,
		],
		dismissText: [text.size.scale(17)],
	}
}

const _invalidScanStyles = StyleSheet.create({
	errorText: {
		fontSize: 14,
		fontFamily: 'Courier',
	},
	body: {
		bottom: 78,
	},
})

const InvalidScanHeader: React.FC<{ title: string }> = ({ title }) => {
	const _styles = useStylesInvalidScan()
	const [{ background, margin, text, border, row, column, color }] = useStyles()

	return (
		<View>
			<View
				style={[
					background.white,
					border.shadow.medium,
					margin.bottom.medium,
					row.item.justify,
					column.justify,
					_styles.header,
				]}
			>
				<Icon
					name='alert-circle-outline'
					width={100}
					height={100}
					fill={color.red}
					style={[row.item.justify]}
				/>
			</View>
			<View>
				<Text style={[text.color.red, text.bold.medium, text.align.center]}>{title}</Text>
			</View>
		</View>
	)
}

const InvalidScanError: React.FC<{ error: string }> = ({ error }) => {
	const [{ border, background, padding, margin, text }] = useStyles()

	return (
		<View style={[border.radius.medium, background.light.red, padding.medium, margin.top.huge]}>
			<Text
				style={[text.color.red, text.align.center, text.bold.medium, _invalidScanStyles.errorText]}
			>
				{error}
			</Text>
		</View>
	)
}

const InvalidScanDismissButton: React.FC<{}> = () => {
	const _styles = useStylesInvalidScan()
	const [{ row, margin, color, padding, text }] = useStyles()
	const navigation = useNavigation()
	const resetDraft = Messenger.useResetDraft()

	return (
		<View style={row.center}>
			<TouchableOpacity
				style={[row.fill, margin.bottom.medium, _styles.dismissButton]}
				onPress={() => {
					resetDraft()
					try {
						navigation.goBack()
					} catch (e) {
						console.warn("couldn't go back:", e)
						navigation.navigate('Tabs')
					}
				}}
			>
				<Icon name='close' width={30} height={30} fill={color.grey} style={row.item.justify} />
				<Text style={[text.color.grey, padding.left.small, row.item.justify, _styles.dismissText]}>
					DISMISS
				</Text>
			</TouchableOpacity>
		</View>
	)
}

const InvalidScan: React.FC<{ title: string; error: string }> = ({ title, error }) => {
	const [layout, setLayout] = useState<number>()
	const [{ background, flex, padding, border }] = useStyles()

	return (
		<View style={[padding.medium, flex.justify.center, { height: '100%' }]}>
			<View
				onLayout={(e) => !layout && setLayout(e.nativeEvent.layout.height)}
				style={[
					background.white,
					padding.medium,
					border.radius.medium,
					layout && { height: layout - 78 },
				]}
			>
				<View style={[_invalidScanStyles.body]}>
					<InvalidScanHeader title={title} />
					<InvalidScanError error={error} />
					<InvalidScanDismissButton />
				</View>
			</View>
		</View>
	)
}

export default InvalidScan
