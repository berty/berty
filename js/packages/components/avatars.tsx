import React from 'react'
import { Image, View, ViewStyle, Text, TouchableOpacity } from 'react-native'
import palette from 'google-palette'
import { SHA3 } from 'sha3'
import { Buffer } from 'buffer'
import { withBadge } from 'react-native-elements'

import { useStyles } from '@berty-tech/styles'
import {
	useAccount,
	useContact,
	useMember,
	useConversation,
	Maybe,
	useMsgrContext,
} from '@berty-tech/store/hooks'
import { navigate } from '@berty-tech/navigation'
import beapi from '@berty-tech/api'
import PinkBotAvatar from '@berty-tech/assets/berty_bot_pink_bg.png'
import GreenDevAvatar from '@berty-tech/assets/berty_dev_green_bg.png'
import OrangeBotAvatar from '@berty-tech/assets/berty_bot_orange_bg.png'
import BlueDevAvatar from '@berty-tech/assets/berty_dev_blue_bg.png'

import AttachmentImage from './AttachmentImage'
import GroupAvatar from './main/Avatar_Group_Copy_19.png'
import Logo from './main/1_berty_picto.svg'

export type AvatarStyle = Omit<
	ViewStyle,
	'borderRadius' | 'width' | 'height' | 'alignItems' | 'justifyContent'
>

const pal = palette('tol-rainbow', 256)

export const GenericAvatar: React.FC<{
	cid: Maybe<string>
	colorSeed: Maybe<string>
	size: number
	style?: AvatarStyle
	isEditable?: boolean
	nameSeed: Maybe<string>
	pressable?: boolean
}> = ({ cid, size, colorSeed, style, isEditable = false, nameSeed, pressable }) => {
	const [{ border, background, color }] = useStyles()
	const padding = Math.round(size / 14)
	let innerSize = Math.round(size - 2 * padding)
	let content: JSX.Element
	if (cid) {
		if (innerSize % 2) {
			innerSize--
		}
		content = (
			<>
				<AttachmentImage
					cid={cid}
					style={{ width: innerSize, height: innerSize, borderRadius: innerSize / 2 }}
					pressable={pressable}
				/>
				{isEditable ? (
					<View
						style={[
							{
								width: innerSize,
								height: innerSize,
								position: 'absolute',
								backgroundColor: color.light.blue,
								opacity: 0.6,
							},
							border.radius.scale(innerSize / 2),
						]}
					/>
				) : null}
			</>
		)
	} else {
		let iconSize = Math.round(innerSize - innerSize / 10) // adjust for jdenticon bug
		if (iconSize % 2) {
			iconSize--
		}
		content = (
			<>
				<NameAvatar size={size} style={style} colorSeed={colorSeed} nameSeed={nameSeed} />
				{isEditable ? (
					<View
						style={[
							{
								width: innerSize,
								height: innerSize,
								position: 'absolute',
								backgroundColor: color.light.blue,
								opacity: 0.6,
							},
							border.radius.scale(innerSize / 2),
						]}
					/>
				) : null}
			</>
		)
	}
	return (
		<>
			<View
				style={[
					background.white,
					border.shadow.medium,
					style,
					{
						borderRadius: size / 2,
						width: size,
						height: size,
						alignItems: 'center',
						justifyContent: 'center',
					},
				]}
			>
				{content}
			</View>
		</>
	)
}

const hardcodedAvatars = {
	berty_dev_green_bg: GreenDevAvatar,
	berty_bot_pink_bg: PinkBotAvatar,
	berty_dev_blue_bg: BlueDevAvatar,
	berty_bot_orange_bg: OrangeBotAvatar,

	group: GroupAvatar,
}

export type HardcodedAvatarKey = keyof typeof hardcodedAvatars

export const HardcodedAvatar: React.FC<{
	size: number
	style?: AvatarStyle
	name: HardcodedAvatarKey
	pressable?: boolean
}> = ({ size, style, name, pressable }) => {
	const [{ border }] = useStyles()
	let avatar = hardcodedAvatars[name]
	if (!avatar) {
		avatar = Logo
	}

	return (
		<TouchableOpacity
			activeOpacity={0.9}
			disabled={!pressable}
			onPress={() => {
				navigate('ImageView', { images: [avatar], previewOnly: true })
			}}
			style={[{ borderRadius: size / 2, backgroundColor: 'white' }, border.shadow.medium, style]}
		>
			<Image
				source={avatar}
				style={{
					width: size,
					height: size,
					borderRadius: size / 2,
				}}
			/>
		</TouchableOpacity>
	)
}

export const AccountAvatar: React.FC<{
	size: number
	style?: AvatarStyle
	isEditable?: boolean
}> = ({ size, style, isEditable }) => {
	const account = useAccount()
	return (
		<GenericAvatar
			nameSeed={account?.displayName}
			cid={account?.avatarCid}
			size={size}
			colorSeed={account?.publicKey}
			style={style}
			isEditable={isEditable}
		/>
	)
}

export const NameAvatar: React.FC<{
	colorSeed: Maybe<string>
	size: number
	style?: AvatarStyle
	nameSeed: Maybe<string>
}> = ({ colorSeed, size, style, nameSeed }) => {
	const h = new SHA3(256).update(Buffer.from(colorSeed || '', 'base64')).digest()
	const color = '#' + pal[h[0]]
	return (
		<View
			style={[
				style,
				{
					width: size,
					height: size,
					backgroundColor: color,
					borderRadius: size / 2,
					alignItems: 'center',
					justifyContent: 'center',
				},
			]}
		>
			<Text style={{ color: 'white', fontSize: size * 0.5, includeFontPadding: false }}>
				{(nameSeed || '?')[0].toUpperCase()}
			</Text>
		</View>
	)
}

export const ContactAvatar: React.FC<{
	publicKey: Maybe<string>
	size: number
	style?: AvatarStyle
	fallbackNameSeed?: Maybe<string>
	pressable?: boolean
}> = ({ publicKey, size, style, fallbackNameSeed, pressable }) => {
	const contact = useContact(publicKey)
	const ctx = useMsgrContext()
	const suggestion = Object.values(ctx.persistentOptions?.suggestions).find(
		(v) => v.pk === publicKey,
	)
	if (suggestion) {
		return (
			<HardcodedAvatar
				size={size}
				style={style}
				name={suggestion.icon as any}
				pressable={pressable}
			/>
		)
	}
	return (
		<GenericAvatar
			nameSeed={contact?.displayName || fallbackNameSeed}
			cid={contact?.avatarCid}
			size={size}
			colorSeed={publicKey}
			style={style}
			pressable={pressable}
		/>
	)
}

export const MemberAvatar: React.FC<{
	publicKey: Maybe<string>
	conversationPublicKey: Maybe<string>
	size: number
	pressable?: boolean
}> = ({ publicKey, conversationPublicKey, size, pressable }) => {
	const member = useMember({ publicKey, conversationPublicKey })
	return (
		<GenericAvatar
			cid={member?.avatarCid}
			size={size}
			colorSeed={publicKey}
			nameSeed={member?.displayName}
			pressable={pressable}
		/>
	)
}

export const MultiMemberAvatar: React.FC<{
	size: number
	style?: AvatarStyle
	publicKey?: Maybe<string>
	fallbackNameSeed?: Maybe<string>
	pressable?: boolean
}> = ({ size, style, publicKey, fallbackNameSeed, pressable }) => {
	const ctx = useMsgrContext()
	const conv = useConversation(publicKey)
	const suggestion = Object.values(ctx.persistentOptions?.suggestions).find(
		(v) => v.pk === publicKey,
	)
	let content: React.ReactElement
	if (suggestion) {
		content = (
			<HardcodedAvatar
				size={size}
				style={style}
				name={suggestion.icon as any}
				pressable={pressable}
			/>
		)
	} else {
		content = (
			<GenericAvatar
				size={size}
				style={style}
				cid={conv?.avatarCid}
				colorSeed={publicKey}
				nameSeed={conv?.displayName || fallbackNameSeed}
				pressable={pressable}
			/>
		)
	}
	const badgeSize = size / 3
	class GroupBadge extends React.Component {
		render = () => <HardcodedAvatar size={badgeSize} name={'group'} />
	}
	const Avatar = () => content
	const WrappedAvatar = withBadge('', { Component: GroupBadge })(Avatar)
	return <WrappedAvatar />
}

export const ConversationAvatar: React.FC<{
	publicKey: Maybe<string>
	size: number
	style?: AvatarStyle
}> = ({ publicKey, size, style }) => {
	const conv = useConversation(publicKey)
	const ctx = useMsgrContext()

	if (conv) {
		if (conv.type === beapi.messenger.Conversation.Type.MultiMemberType) {
			return <MultiMemberAvatar size={size} style={style} publicKey={publicKey} />
		} else if (conv.type === beapi.messenger.Conversation.Type.ContactType) {
			return <ContactAvatar size={size} publicKey={conv?.contactPublicKey} />
		}
	}

	const suggestion = Object.values(ctx.persistentOptions?.suggestions).find(
		(v) => v.pk === publicKey,
	)
	if (suggestion) {
		return <HardcodedAvatar size={size} style={style} name={suggestion.icon as any} />
	}

	return <GenericAvatar size={size} style={style} cid='' colorSeed={publicKey} nameSeed={'C'} />
}
