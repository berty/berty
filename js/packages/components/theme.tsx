import React from 'react'
import { ApplicationProvider } from '@ui-kitten/components'
import { mapping, light } from '@eva-design/eva'
import { StatusBarWrapper } from './common'

export const Provider: React.FC = ({ children }) => (
	<>
		<ApplicationProvider
			mapping={mapping}
			customMapping={
				{
					strict: {
						'text-font-family': 'Open Sans',
						'text-paragraph-1-font-weight': '600',
					},
				} as any
			}
			theme={light}
		>
			<StatusBarWrapper />
			{children}
		</ApplicationProvider>
	</>
)
