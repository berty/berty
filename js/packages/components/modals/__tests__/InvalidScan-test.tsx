import React from 'react'
import renderer from 'react-test-renderer'
import { NavigationContainer } from '@react-navigation/native'
import { IconRegistry } from '@ui-kitten/components'
import { EvaIconsPack } from '@ui-kitten/eva-icons'
import { SafeAreaProvider } from 'react-native-safe-area-context'

import { Provider as ThemeProvider } from '@berty-tech/components/theme'
import { FeatherIconsPack } from '@berty-tech/messenger-app/feather-icons'
import { CustomIconsPack } from '@berty-tech/messenger-app/custom-icons'
import { Provider as StyleProvider } from '@berty-tech/styles'

import InvalidScan from '../InvalidScan'

jest.mock('react-native/Libraries/LogBox/LogBox')

describe('InvalidScan modal', () => {
	beforeEach(() => {
		jest.resetModules()
		jest.resetAllMocks()
		jest.useFakeTimers()
	})
	it('renders correctly', () => {
		const TestProvider: React.FC = ({ children }) => (
			<SafeAreaProvider
				initialMetrics={{
					frame: { x: 0, y: 0, width: 0, height: 0 },
					insets: { top: 0, left: 0, right: 0, bottom: 0 },
				}}
			>
				<StyleProvider>
					<IconRegistry icons={[EvaIconsPack, FeatherIconsPack, CustomIconsPack]} />
					<ThemeProvider>
						<NavigationContainer>{children}</NavigationContainer>
					</ThemeProvider>
				</StyleProvider>
			</SafeAreaProvider>
		)
		const props: React.ComponentProps<typeof InvalidScan> = {
			type: 'link',
			error: new Error('Test error'),
		}
		const tree = renderer
			.create(
				<TestProvider>
					<InvalidScan {...props} />
				</TestProvider>,
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
