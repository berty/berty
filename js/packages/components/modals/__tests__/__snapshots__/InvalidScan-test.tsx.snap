// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvalidScan modal renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "padding": 16,
          },
          Object {
            "height": "100%",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "padding": 16,
            },
            Object {
              "borderRadius": 16,
            },
            Object {
              "backgroundColor": "#FFFFFF",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "bottom": 78,
              },
            ]
          }
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 2,
                  },
                  Object {
                    "marginBottom": 16,
                  },
                  Object {
                    "alignSelf": "center",
                  },
                  Object {
                    "alignItems": "stretch",
                    "flexDirection": "column",
                    "justifyContent": "space-evenly",
                  },
                  Array [
                    Object {
                      "width": 120,
                    },
                    Object {
                      "height": 120,
                    },
                    Object {
                      "borderRadius": 60,
                    },
                  ],
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "shadowColor": "#000000",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={100}
                  bbWidth={100}
                  fill="#F64278"
                  focusable={false}
                  height={100}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Array [
                        Object {
                          "alignSelf": "center",
                        },
                      ],
                      Object {
                        "flex": 0,
                        "height": 100,
                        "width": 100,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                  width={100}
                >
                  <RNSVGGroup
                    fill={4294328952}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGRect
                          height="24"
                          opacity={0}
                          width="24"
                          x={0}
                          y={0}
                        />
                        <RNSVGPath
                          d="M12 2a10 10 0 1 0 10 10A10 10 0 0 0 12 2zm0 18a8 8 0 1 1 8-8 8 8 0 0 1-8 8z"
                        />
                        <RNSVGCircle
                          cx="12"
                          cy="16"
                          r="1"
                        />
                        <RNSVGPath
                          d="M12 7a1 1 0 0 0-1 1v5a1 1 0 0 0 2 0V8a1 1 0 0 0-1-1z"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
            <View>
              <Text
                appearance="default"
                category="p1"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "Open Sans",
                      "fontSize": 15,
                      "fontWeight": "600",
                    },
                    Array [
                      Object {
                        "fontWeight": "bold",
                      },
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#F64278",
                      },
                    ],
                  ]
                }
              >
                modals.manage-deep-link.invalid-link
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderRadius": 16,
                },
                Object {
                  "padding": 16,
                },
                Object {
                  "marginTop": 40,
                },
                Object {
                  "backgroundColor": "#F6427840",
                },
              ]
            }
          >
            <Text
              appearance="default"
              category="p1"
              style={
                Array [
                  Object {
                    "color": "#222B45",
                    "fontFamily": "Open Sans",
                    "fontSize": 15,
                    "fontWeight": "600",
                  },
                  Array [
                    Object {
                      "textAlign": "center",
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                    Object {
                      "fontFamily": "Courier",
                      "fontSize": 14,
                    },
                    Object {
                      "color": "#F64278",
                    },
                  ],
                ]
              }
            >
              Error: Test error
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "stretch",
                "flexDirection": "row",
                "justifyContent": "space-evenly",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "stretch",
                  "borderColor": "#A5A7C5",
                  "borderRadius": 9,
                  "borderWidth": 2,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 16,
                  "marginTop": 50,
                  "opacity": 1,
                  "paddingBottom": 9,
                  "paddingLeft": 9,
                  "paddingRight": 16,
                  "paddingTop": 9,
                }
              }
            >
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={30}
                  bbWidth={30}
                  fill="#979797"
                  focusable={false}
                  height={30}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "alignSelf": "center",
                      },
                      Object {
                        "flex": 0,
                        "height": 30,
                        "width": 30,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                  width={30}
                >
                  <RNSVGGroup
                    fill={4288124823}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGRect
                          height="24"
                          matrix={
                            Array [
                              -1,
                              1.2246467991473532e-16,
                              -1.2246467991473532e-16,
                              -1,
                              24,
                              24,
                            ]
                          }
                          opacity={0}
                          width="24"
                          x={0}
                          y={0}
                        />
                        <RNSVGPath
                          d="M13.41 12l4.3-4.29a1 1 0 1 0-1.42-1.42L12 10.59l-4.29-4.3a1 1 0 0 0-1.42 1.42l4.3 4.29-4.3 4.29a1 1 0 0 0 0 1.42 1 1 0 0 0 1.42 0l4.29-4.3 4.29 4.3a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42z"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <Text
                appearance="default"
                category="p1"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "Open Sans",
                      "fontSize": 15,
                      "fontWeight": "600",
                    },
                    Array [
                      Object {
                        "paddingLeft": 9,
                      },
                      Object {
                        "alignSelf": "center",
                      },
                      Array [
                        Object {
                          "fontSize": 34,
                        },
                      ],
                      Object {
                        "color": "#979797",
                      },
                    ],
                  ]
                }
              >
                modals.invalid-scan.dismiss-button
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
